go/vt/vttablet/tabletmanager/rpc_external_reparent.go

```diff
@@ -160,9 +161,14 @@ func (agent *ActionAgent) finalizeTabletExternallyReparented(ctx context.Context
 		}
 	}()
 
+	tablet := agent.Tablet()
+	tabletMap, err := agent.TopoServer.GetTabletMapForShard(ctx, tablet.Keyspace, tablet.Shard)
```

Comment:
do you mean print to the log or collect it in `errs`?

Reply:
I think it needs to be printed to log and not collected in `errs`. If we collect it in `errs` it will cause us to bail out later. However, that will mean TER gets stuck/blocked if any topo cells are unavailable.

This is a common concern throughout Vitess. Any time we try to access topo cross-cell (as we're doing here, since GetTabletMapForShard has to read tablets from all cells where the shard has tablets), we need to be careful to not get blocked in the case when one or more cells is unavailable. Otherwise, every cell becomes a single point of failure for that operation.

Reply:
I suppose this is one of the reasons we call finalize with a relatively short timeout (30s), because any of the various things we are doing in finalize could block if the target component is unavailable.